### ISSUE TITLE
ERROR: type FourRooms has no field agent_dir

### ISSUE THREAD

**Issue by Sid-Bhatia-0** [JuliaReinforcementLearning/Gridworld.jl#14]
```
julia> w3 = FourRooms()

julia> w3(MOVE_FORWARD)
ERROR: type FourRooms has no field agent_dir
Stacktrace:
 [1] getproperty(::FourRooms, ::Symbol) at ./Base.jl:33
 [2] (::FourRooms)(::Gridworld.MoveForward) at /home/sid/.julia/packages/Gridworld/yQBUD/src/envs/fourrooms.jl:25
 [3] top-level scope at REPL[10]:1
```

`FourRooms` functor tries to access `w.agent_dir`, a field that does not exist. In file [fourrooms.jl](https://github.com/JuliaReinforcementLearning/Gridworld.jl/blob/master/src/envs/fourrooms.jl)

```
mutable struct FourRooms <: AbstractGridWorld
    world::GridWorldBase{Tuple{Empty,Wall,Goal}}
    agent_pos::CartesianIndex{2}
    agent::Agent
end
```

```
function (w::FourRooms)(::MoveForward)
    dest = w.agent_dir(w.agent_pos)
    if !w.world[WALL, dest]
        w.agent_pos = dest
    end
    w
end
```

**Comment by Sid-Bhatia-0** [comment]
This has been fixed in 58c071b8756731a32f460f33dcaa8e9e0d62d5c3

**Comment by findmyway** [comment]
Yeah, I realized this problem while generating the example GIF in README.md and then had it fixed.
